connector/connect/common/src/main/protobuf/spark/connect/commands.proto

```diff
@@ -209,6 +209,15 @@ message WriteStreamOperationStart {
     string path = 11;
     string table_name = 12;
   }
+
+  optional Foreach foreach = 13;
+}
+
+message Foreach {
+  // (Required) The encoded commands of the Python foreach function
+  bytes command = 1;
```

Comment:
Can we use `PythonUDF` message? Even though 'output_type' is not needed.

Reply:
Right we can reuse. But I think foreach conceptually doesn't belong to `PythonUDF`. We probably need to refactor this, maybe in a follow-up PR? I'm thinking that there probably needs to be a more general `message PythonFunction` that wraps all info, and `PythonUDF` could just wrap around it and here we could just use it. WDYT?

Reply:
@amaliujia Hi Rui, I'm not an expert on protobuf... Can I get some input from you? Thanks in advance!

Reply:
Feels we can reuse PythonUDF, then you don't `transformPythonForeachFunction` on server, which is duplicate or existing one?

Reply:
What breaks if we do reuse it? Given there will more updates to PythonUDF (libraries etc), reusing would reduce work in the future.

Reply:
Nothing actually breaks if we reuse it. And I'm fine with that if people think that's the correct way. 

But I really think it's not semantically right to call `foreach` function a python UDF. They are not the same. And `foreachWriter` doesn't require `output_type` and `eval_type` to be set.

A better way is to create a new proto message called `pythonFunction` that contains the `byte command` and `string python_ver`. And have `PythonUDF` include it as a member. And here we can just reuse `pythonFunction`. 

But I'm not expert on protoBuf I'm not sure if this is a breaking change. 

@HyukjinKwon @amaliujia Can we get some input from you guys? Thank you!!